src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -25,6 +25,8 @@
 // STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
 // THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+#ifndef APPLE_DEBUG_BLOCKCHAIN_DB
```

Comment:
Yes it does, but only when the compiler is building libblockchain_db. The same db_lmdb.cpp file gets compiled again when building libcryptonote_core, and this time the whole source code gets compiled.